### PR TITLE
Carlos Volunteer Hours Distribution

### PR DESCRIPTION
**Pull request by cgomezhub**
# Description
This part of a new Report Page: Create Company Summary Dashboard for Admins requested by Jae.
Implements # (WBS) 
Volunteer Hours Distriibution module

Screenshot 2024-07-20 162442

Screenshot 2024-07-20 162549

Screenshot 2024-07-20 162605

## Backend branch to use:
- development branch

## Features added:
-  HoursWorkedPieChart component
- HoursWorkedList component
- NumbersVolunteerWorked  component
…

## How to test:
1. Check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to dashboard→ Report→ Total Org Summary
6. verify pieChart and info labels (see pictures and compare)

## videos of change
General view - setting up test dates:
https://www.loom.com/share/88250d847a974950854e152abd12de88?sid=eecd1398-1d8d-46dc-8262-53285eb00460

Dark Mode Functionality:
https://www.loom.com/share/f09eb36a755346c7b7a6e18a834e1e5a?sid=f9aeaebf-1ab8-4d60-a232-4a562e859898

## Note:
- This module currently compares the last week to the previous week. You can manually change the date settings on your PC and restart the Frontend App (see the video above). Soon, It will be connected to the Select Date Range and Comparison Period module, which is under development (see below picture)

Screenshot 2024-07-20 164233

### CODE REVIEW

**Review by SrikanthChowdary495 — COMMENTED** [review]
I have tested this PR. This module currently compares the current week to the previous week. I have manually changed the PC's date settings and restarted the Frontend App as shown in the picture. soon, It will be connected to the Select Date Range and Comparison Period module.  

I was able to see the changes in the different dates in Volunteer Hours Distribution. Attaching the reference images.

**Review by Chetan0606 — APPROVED** [review]
Verified the PR and everything looks good and I was able to observe the change in the hours. Please find the below attached images
Screenshot (335)
Screenshot (337)

**Comment by souravwalke** [comment]
Verified the PR and everything is working as expected. 

PR#2503

File: src/components/Header/Header.jsx [hunk 1]
```diff
@@ -333,6 +353,11 @@ export function Header(props) {
                           {WEEKLY_SUMMARIES_REPORT}
                         </DropdownItem>
                       )}
+                      {canGetWeeklyVolunteerSummary && (
+                        <DropdownItem tag={Link} to="/totalorgsummary">
```
Comment (kabraambika) [inline src/components/Header/Header.jsx:357]: In dark mode:

1. text color of menu item is black which is not correct as shown below:
image

2. Hover over the menu item, highlight color is different and not consistent for new menu item:
**Color of highlight for new menu item is:** 
image

**Color of highlight for existing menu item is:** 
image

File: src/components/TotalOrgSummary/TotalOrgSummary.jsx [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+                    <div className="d-flex flex-column align-items-center justify-content-center">
+                      <HoursWorkList darkMode={darkMode} usersTimeEntries={usersTimeEntries} />
+                      <NumbersVolunteerWorked usersTimeEntries={usersTimeEntries} />
+                    </div>
+                  </>
+                )}
+              </div>
+            </div>
+          </Col>
+          <Col lg={{ size: 3 }}>
```
Comment (kabraambika) [inline src/components/TotalOrgSummary/TotalOrgSummary.jsx:260]: In dark mode, text is not visible as shown below:

image

**Review by kabraambika — COMMENTED** [review]
I have reviewed the Volunteer Workload section and provided a few comments for your consideration.

**Review by ReinaT5678 — DISMISSED** [review]
The functionality seems to be working well for me! I showed my testing in the video below:

https://github.com/user-attachments/assets/e4536875-8618-434e-9de3-64961630d448

I also noticed there are some inconsistencies in dark mode such as the loading animation and the text being difficult to see. 
Screenshot 2024-07-15 at 1 52 16 PM

**Review by nikhilpittala16 — APPROVED** [review]
I have tested the PR and everything works as expected. 
Screenshot (149)

**Review by vishavk1992 — DISMISSED** [review]
I have tested this PR and it works as expected in light mode, but some changes are needed for dark mode.
Please check screenshots 
PR-2503

Changes require for Dark Mode: 
PR-2503-1

**Comment by nikethaanand** [comment]
The changes are working, however the task Completed and hours completed are not visible in dark mode. Please do change the font.

image

**Review by jinxyou — DISMISSED** [review]
I tested this PR and the pie chart works as expected. Great work!
image

**Review by Gauravmg6 — APPROVED** [review]
Everything works as expected

Screenshot 2024-07-28 003459

**Comment by cgomezhub** [comment]
> I have tested this PR and it works as expected in light mode, but some changes are needed for dark mode. 

please check again and review it. Thanks

File: src/components/TotalOrgSummary/HoursWorkList/HoursWorkList.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      { name: '20 - 29.99', min: 20, max: 29.99, color: '#FFBB28' },
+      { name: '30 - 39.99', min: 30, max: 39.99, color: '#FF8042' },
+      { name: '40+', min: 40, max: Infinity, color: '#800080' },
+    ];
+
+    const arrData = ranges.map(range => {
+      const value = usersTimeEntries.reduce((acc, curr) => {
+        const hours = Number(curr.hours) || 0;
+        const minutes = Number(curr.minutes) || 0;
+        const total = hours + minutes / 60;
```
[the remark this answers is not in the document]
Comment (cgomezhub) [reply src/components/TotalOrgSummary/HoursWorkList/HoursWorkList.jsx:23]: thanks for your suggestion, nevertheless the figma design request is with out decimal, please see pictures above.

File: src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.jsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import React, { useEffect, useState } from 'react';
-import { ProjectPieChart } from '../ProjectPieChart/ProjectPieChart';
+import {ProjectPieChart} from "../ProjectPieChart/ProjectPieChart";
```
[the remark this answers is not in the document]
Comment (cgomezhub) [reply src/components/Reports/ProjectReport/PiechartByProject/PieChartByProject.jsx:2]: it is done!. plesase review  and  approve the PR as soon as you can. Thanks

File: src/components/TotalOrgSummary/HoursWorkList/HoursWorkList.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kabraambika) [reply src/components/TotalOrgSummary/HoursWorkList/HoursWorkList.jsx:23]: Got it, Thanks!

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (kabraambika) [reply src/components/Header/Header.jsx:357]: Hello @cgomezhub , This observation is still occurring after latest changes

File: src/components/TotalOrgSummary/TotalOrgSummary.jsx (hunk 2 above)
Comment (cgomezhub) [reply src/components/TotalOrgSummary/TotalOrgSummary.jsx:268]: This is not part of this PR, it is only Volunteer Hours Distribution

Comment (kabraambika) [reply src/components/TotalOrgSummary/TotalOrgSummary.jsx:268]: I have a question regarding the Volunteer workload display. When the card is collapsed, it appears in dark mode as shown below:
image

However, when expanded, the header of the card appears white while the chart and other elements remain in dark mode, as shown below:
image

Is this the expected UI in dark mode? 
Additionally, would it be possible to change the text inside the pie chart ("HOURS WORKED") and the labels to white color in dark mode?

Thank you!

File: src/components/Header/Header.jsx (hunk 1 above)
Comment (kabraambika) [reply src/components/Header/Header.jsx]: Resolving the conversation, working fine after latest changes in [1a2132e](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2503/commits/1a2132e0146cf41168d15756c0587fb267d42ee5)

Comment (cgomezhub) [reply src/components/Header/Header.jsx]: it's done. Thanks!

**Review by kabraambika — APPROVED** [review]
I have reviewed the frontend PR after latest changes with the development branch in the backend. Everything is working fine in both light and dark modes. Thank you for addressing my questions.

**Review by Chetan0606 — DISMISSED** [review]
Verified the PR as per the provided description and checked it in both light and dark modes and it looks good.
image

if you agree, please check approve here:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2503/files/8d6700a46c760d404033b0d92dca53b5c9ac07c9

**Comment by arunsrao9** [comment]
Verified the PR as per the provided description and checked it in both light and dark modes.
Light mode:
Screenshot 2024-07-30 at 11 52 57 AM

Dark mode:(There is a little difficulty in reading the text, inside the pie, when in dark mode)
Screenshot 2024-07-30 at 11 52 50 AM
Looks good to me, Great work!

**Review by varunreddy03 — DISMISSED** [review]
I have checked into the current branch and verified the PR 2503,
Case 1: For the current Date in both light and dark mode the Pie chart, the number of hours, and the number of volunteers worked all the 3 features are working as intended.

PR 2503_1
PR 2503_4


Case 2: Then I have manually changed the system time and restarted the frontend and checked for other dates in both light and dark theme and it is working good for all 3 parameters.

PR 2503_2
PR 2503_3

**Comment by beblicarl** [comment]
This functionality works as intended on both light mode and dark mode and the code is great

image

image

**Review by beblicarl — APPROVED** [review]

**Review by ReinaT5678 — DISMISSED** [review]
The changes look great! Everything is readable in dark mode.

Screenshot 2024-07-31 at 4 27 29 PM

Screenshot 2024-07-31 at 4 27 47 PM

**Review by Pranay-ai — COMMENTED** [review]
Hey Carlos I have  tested the PR and ONE OF THE CHART IS DISPLAYING AS EXPECTED BUT THE OTHER CHARTS ARE HAVING A BACKEND ISSUE
image
Thank You

**Review by Pranay-ai — DISMISSED** [review]
Hey @cgomezhub 
The PR Is working as expected.
Thank You

**Review by vishavk1992 — DISMISSED** [review]
I have tested this PR again now it works as expected for dark mode also 

Screenshot 2024-08-07 134403

**Review by AnkitLall — APPROVED** [review]
Hi @cgomezhub,

I have tested the feature, following the instructions in the PR, and it works as expected. Refer to the screenshots below.

2024-08-01 at 12 55 39 AM
2024-08-09 at 12 55 20 AM

**Review by aaryaneil — DISMISSED** [review]
Tested the PR and the changes work as expected.
Screenshot 2024-08-10 at 8 02 28 PM
Screenshot 2024-08-01 at 8 03 06 PM

**Review by arunsrao9 — DISMISSED** [review]
Verified the PR as per the provided description and checked it in both light and dark modes.
Looks good!

**Review by ReinaT5678 — APPROVED** [review]

**Review by nikhilpittala16 — APPROVED** [review]

**Review by preethamdnr — APPROVED** [review]
Hello Carlos,

I have tested this PR, working as expected in both the modes.
image
image

**Review by SnelPatare — COMMENTED** [review]
I tested this PR and it is working well.. 
I few suggestion for feature:

1.  Please change "task Completed" to "task completed". Case is not uniform. Image is as below.
image

2.  0% indicates no information in piechart. It will be better to remove them from the chart.  Image is as below
image

3.  I ran the applicaion in dark mode and I could see different in tab color.  Rest all tabs are white and "Volunteer Worload and Task Completion Analysis" is in black. It is misleading as user because it seems that Option "Volunteer Workload and Task Completion Analysis" is disabled. When we click (open) on " Volunteer Workload and Task Completion Analysis" it changes its color to white and when we close it changes to black again. This is not seen in light mode.  Images as below

Dark Mode Images:
image

image

Light mode Images:
image

image

**Review by adityasure9 — APPROVED** [review]
This PR successfully implements the Volunteer Hours Distribution feature with the expected functionality. 

image

**Comment by cgomezhub** [comment]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2503/#pullrequestreview-2243231755

It have been addressed, please review it 

fix-0%-CapitalLetter


Screenshot 2024-08-19 152919

**Comment by one-community** [comment]
Thank you all, merging!

**Comment by SnelPatare** [comment]
Working perfectly !!

**Pull request merged**